vmx/src/vm.rs

```diff
@@ -16,28 +22,41 @@ use crate::module::WrappedModule;
 use crate::session::{Session, SessionMut};
 use crate::types::{Error, StandardBufSerializer};
 
-#[derive(Clone, Copy, PartialOrd, Ord, PartialEq, Eq)]
-pub struct ModuleId(usize);
+#[derive(Debug, Clone, Copy, PartialOrd, Ord, PartialEq, Eq)]
+pub struct CommitId(usize);
 
 #[derive(Default)]
 pub struct VM {
     modules: BTreeMap<ModuleId, WrappedModule>,
+    hosted_queries: NativeQueries,
 }
```

Comment:
Personally, I think the `VM` struct should be nothing but something like a file lock signifying mutable access to some directory where we structure the module diffs and all other data we might need. Keeping in mind that only the `VM` struct should **write** to disk, this allows us to use Rust's borrow checker to prevent issues like the "interceding update problem".